src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -2469,6 +2470,10 @@ public override BoundNode VisitUsingStatement(BoundUsingStatement node)
                 VisitStatement(node.DeclarationsOpt);
             }
 
+            if (((UsingStatementSyntax)node.Syntax).AwaitKeyword != default)
+            {
+                _pendingBranches.Add(new PendingBranch(node, this.State));
```

Comment:
>_pendingBranches.Add(new PendingBranch(node, this.State)); [](start = 16, length = 58)

When exactly should this be done and why? For example, should we do this after the Body is visited? We will be awaiting after the body (ignoring exception scenarios), so should the State reflect that?